ADR/0044-spdx-support.md

```diff
@@ -0,0 +1,279 @@
+We run syft/cachi2 on source directory of a project which should be build into container. Generated SBOM contains
+```json
+{
+  "SPDXID": "SPDXRef-DOCUMENT",
+  ...
+  "packages": [
+      {
+        "name": ".",
+        "SPDXID": "SPDXRef-DocumentRoot-Directory-.",
+        "supplier": "NOASSERTION",
+        "downloadLocation": "NOASSERTION",
+        "filesAnalyzed": false,
+        "licenseConcluded": "NOASSERTION",
+        "licenseDeclared": "NOASSERTION",
+        "primaryPackagePurpose": "FILE"
+      },
+      {
+      {
+        "name": "attrs",
+        "SPDXID": "SPDXRef-Package-python-attrs-eef51168ca2a575f",
+        "versionInfo": "24.2.0",
+        ...
+      }
+      ...
+  ],
+  "relationships": [
+      {
+        "spdxElementId": "SPDXRef-DOCUMENT",
+        "relatedSpdxElement": "SPDXRef-DocumentRoot-Directory-.",
+        "relationshipType": "DESCRIBES"
+      },
+      {
+          "spdxElementId": "SPDXRef-DocumentRoot-Directory-.",
+          "relatedSpdxElement": "SPDXRef-Package-python-attrs-eef51168ca2a575f",
+          "relationshipType": "CONTAINS"
+      }
+      ...
+  ]
+}
+```
+
+And we want that to express that SBOM is actually generated for a container image not for source directory.
+So we remove `SPDXRef-DocumentRoot-Directory-.` package and add new virtual package representing the container image. And replace SPDX ID in relationships with ID of the new package. New SBOM should look like this:
+
+```json
+{
+  "SPDXID": "SPDXRef-DOCUMENT",
+  ...
+  "packages": [
+      {
+        "name": "container-registry.com/my-org/my-image:latest",
+        "SPDXID": "SPDXRef-DocumentRoot-Image-container-registry.com/my-org/my-image:latest",
+        "versionInfo": "sha256:9ac75c1a392429b4a087971cdf9190ec42a854a169b6835bc9e25eecaf851258",
+        ...
+        "checksums": [
+            {
+                "algorithm": "SHA256",
+                "checksumValue": "9ac75c1a392429b4a087971cdf9190ec42a854a169b6835bc9e25eecaf851258"
+            }
+        ],
+        ...
+        "externalRefs": [
+            {
+                "referenceCategory": "PACKAGE-MANAGER",
+                "referenceType": "purl",
+                "referenceLocator": "pkg:oci/container-registry.com/my-org/my-image@sha256:9ac75c1a392429b4a087971cdf9190ec42a854a169b6835bc9e25eecaf851258"
```

Comment:
In the interest of aligning with https://github.com/konflux-ci/build-tasks-dockerfiles/pull/181, I think these should be 
```suggestion
      {
        "name": "my-image",
        "SPDXID": "SPDXRef-image",
        "versionInfo": "latest",
        ...
        "checksums": [
            {
                "algorithm": "SHA-256",
                "checksumValue": "9ac75c1a392429b4a087971cdf9190ec42a854a169b6835bc9e25eecaf851258"
            }
        ],
        ...
        "externalRefs": [
            {
                "referenceCategory": "PACKAGE-MANAGER",
                "referenceType": "purl",
                "referenceLocator": "pkg:oci/my-image@sha256:9ac75c1a392429b4a087971cdf9190ec42a854a169b6835bc9e25eecaf851258?repository_url=container-registry.com/my-org/my-image"
```

Reply:
It's just example, it doesn't really have to be 1:1 with real world output, but ok